contrib/experimental/great_expectations_experimental/expectations/expect_yesterday_count_compared_to_avg_equivalent_days_of_week.py

```diff
@@ -0,0 +1,217 @@
+                    "in": {
+                        "column": "column_b",
+                        "run_date": TODAY_STR,
+                    },
+                    "out": {"success": False},
+                },
+            ],
+            "test_backends": [
+                {
+                    "backend": "sqlalchemy",
+                    "dialects": ["sqlite"],
+                }
+            ],
+        }
+    ]
+
+    metric_dependencies = ("column.counts_per_days_custom",)
+    success_keys = (
+        "run_date",
+        "threshold",
+    )
+
+    def validate_configuration(
+        self, configuration: Optional[ExpectationConfiguration]
+    ) -> None:
+        # Setting up a configuration
+        super().validate_configuration(configuration)
+
+    def _validate(
+        self,
+        configuration: ExpectationConfiguration,
+        metrics: Dict,
+        runtime_configuration: dict = None,
+        execution_engine: ExecutionEngine = None,
+    ):
+        date_format = "%Y-%m-%d"
+        from datetime import datetime, timedelta
+
+        counts_per_days = metrics["column.counts_per_days_custom"]
+        run_date: str = self.get_success_kwargs(configuration).get("run_date")
+        threshold: float = float(
+            self.get_success_kwargs(configuration).get("threshold")
+        )
+        run_date_as_date: datetime = datetime.strptime(run_date, date_format)
+        equivalent_days_in_previous_weeks: List[datetime] = [
+            run_date_as_date - timedelta(delta) for delta in [7, 14, 21, 28]
+        ]
+        equivalent_days_in_previous_weeks_str: List[str] = [
+            datetime.strftime(i, date_format) for i in equivalent_days_in_previous_weeks
+        ]
+
+        previous_days_counts: List[int] = [
+            i[1]
+            for i in counts_per_days
+            if i[0] in equivalent_days_in_previous_weeks_str
+        ]
+        yesterday_count: int = [i[1] for i in counts_per_days if i[0] == run_date][0]
+
+        avg_equivalent_previous_days_count = 0
+        if len(previous_days_counts) > 0:
+            avg_equivalent_previous_days_count = sum(previous_days_counts) / len(
+                previous_days_counts
+            )
+
+        absolute_diff = abs(yesterday_count - avg_equivalent_previous_days_count)
+        diff_percentage = (
+            absolute_diff / avg_equivalent_previous_days_count
+            if avg_equivalent_previous_days_count > 0
+            else 1
+        )
+
+        if diff_percentage > threshold:
+            msg = (
+                f"The diff between yesterday's count and the avg. count ({diff_percentage:.0%}) exceeds the defined "
+                f"threshold ({threshold:.0%})"
+            )
+            success = False
+        else:
+            msg = (
+                f"The diff between yesterday's count ({yesterday_count}) and the avg. count ({diff_percentage:.0%}) "
+                f""
+                f""
```

Comment:
Just a nitpick, would you mind removing these two extra `f""` lines?

Reply:
Sure